app/grandchallenge/algorithms/serializers.py

```diff
@@ -96,6 +96,12 @@ class JobSerializer(serializers.ModelSerializer):
         read_only=True,
         allow_null=True,
     )
+    optional_hanging_protocols = HangingProtocolSerializer(
+        many=True,
+        source="algorithm_image.algorithm.optional_hanging_protocols",
```

Comment:
We need to ensure that this does not introduce any N+1 queries. Inspect the SQL calls with the django debug toolbar, or directly with shell plus. Ensure that the API list and detail views are not having to do multiple database requests when multiple objects have multiple optional hanging protocols. If so, select related or prefetch related statements may need to be added to the queryset in the view.

Same for the other apps.